manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -168,9 +168,13 @@ template: |
       value: {{ .Values.global.jwtPolicy }}
     - name: PILOT_CERT_PROVIDER
       value: {{ .Values.global.pilotCertProvider }}
-    {{- if .Values.global.caAddress }}
     - name: CA_ADDR
+    {{- if .Values.global.caAddress }}
       value: {{ .Values.global.caAddress }}
+    {{- else if and .Values.global.remotePilotAddress .Values.pilot.enabled }}
```

Comment:
would not the value be line 175 regardless of whether pilot is enabled?   When pilot is not enabled, we set up the istiod endpoint to point to remote istiod.

Reply:
@linsun I thought that `.Values.pilot.enabled` was `false` with "central istiod" ... is that not the case? Basically what I'm trying to do here is to only specify `CA_ADDR` when we're using the legacy remote cluster. Once we completely replace the legacy remote in favor of the new remote (central istiod) then none of this will be needed since there will be no `istiod-remote` service confusing things.

Reply:
That is correct, central istiod is when pilot is disabled on remote cluster.   Your new change looks good to me.